Source/WebCore/accessibility/cocoa/AXCoreObjectCocoa.mm

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "config.h"
+#import "AXCoreObject.h"
+
+#import "ColorCocoa.h"
+#import "WebAccessibilityObjectWrapperBase.h"
+
+#if PLATFORM(COCOA)
+
+namespace WebCore {
+
+// When modifying attributed strings, the range can come from a source which may provide faulty information (e.g. the spell checker).
+// To protect against such cases, the range should be validated before adding or removing attributes.
+bool attributedStringContainsRange(NSAttributedString *attributedString, const NSRange& range)
+{
+    return NSMaxRange(range) <= attributedString.length;
+}
+
+void attributedStringSetFont(NSMutableAttributedString *attributedString, CTFontRef font, const NSRange& range)
+{
+    if (!attributedStringContainsRange(attributedString, range) || !font)
+        return;
+
+    auto fontAttributes = adoptNS([[NSMutableDictionary alloc] init]);
+    auto familyName = adoptCF(CTFontCopyFamilyName(font));
+    NSNumber *size = [NSNumber numberWithFloat:CTFontGetSize(font)];
+
+#if PLATFORM(IOS_FAMILY)
+    auto fullName = adoptCF(CTFontCopyFullName(font));
+    if (fullName)
+        [fontAttributes setValue:bridge_cast(fullName.get()) forKey:AccessibilityTokenFontName];
+    if (familyName)
+        [fontAttributes setValue:bridge_cast(familyName.get()) forKey:AccessibilityTokenFontFamily];
+    if ([size boolValue])
```

Comment:
this is a NSNumber wrapping a float. So boolValue might not be correct

Reply:
I think it's checking whether the `NSNumber` is greater than zero (but maybe `boolValue` doesn't work this way, I'm not sure) 🤔 and if so, we set the font size with the actual NSNumber value:

```objc
if ([size boolValue])
    [fontAttributes setValue:size forKey:AccessibilityTokenFontSize];
```

But I just moved this function from elsewhere, didn't change it at all. If there's a better way to do this I'm happy to change it though, just let me know what you think we should do!